miso-web/src/spring-test/java/uk/ac/bbsrc/tgac/miso/webapp/springtest/HotRestControllerST.java

```diff
@@ -0,0 +1,111 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.*;
+import org.springframework.test.web.servlet.ResultActions;
+import com.jayway.jsonpath.JsonPath;
+
+import static org.hamcrest.Matchers.*;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+import org.springframework.test.web.servlet.MvcResult;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.web.servlet.View;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.springframework.security.test.context.support.WithMockUser;
+import uk.ac.bbsrc.tgac.miso.core.data.type.StatusType;
+import static org.junit.Assert.*;
+import java.util.Collections;
+import uk.ac.bbsrc.tgac.miso.webapp.controller.rest.HotRestController.SpreadsheetDataDto;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import org.springframework.test.web.servlet.MockMvc;
+import java.util.Date;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+
+public class HotRestControllerST extends AbstractST {
+
+  private static final String CONTROLLER_BASE = "/rest/hot";
+
+  @Test
+  public void testDownloadSpreadsheet() throws Exception {
+    SpreadsheetDataDto sheet = new SpreadsheetDataDto();
+
+    List<String> headers = Arrays.asList("1", "2", "3", "4", "5");
+    List<String> firstrow = Arrays.asList("one", "two", "three", "four", "five");
+    List<String> secondrow = Arrays.asList("un", "deux", "trois", "quatre", "cinq");
+    List<List<String>> rows = Arrays.asList(firstrow, secondrow);
+
+    sheet.setHeaders(headers);
+    sheet.setRows(rows);
+
+    MockHttpServletResponse response = getMockMvc()
+        .perform(post(CONTROLLER_BASE + "/spreadsheet").param("format", "CSV").content(makeJson(sheet))
+            .contentType(MediaType.APPLICATION_JSON))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType("text/csv"))
+        .andExpect(header().longValue("Content-Length", 83)).andReturn().getResponse();
+    String[][] records = new String[headers.size()][rows.size() + 1];
+    String[] rawRows = response.getContentAsString().split("\r");
+
+    for (int i = 0; i < rawRows.length; i++) {
+      String s = rawRows[i].replaceAll("\\r", "");
+      s = s.replaceAll("\\n", "");
+      s = s.replaceAll("\"", "");
+      records[i] = s.split(",");
+    }
+    checkArray(records[0], headers);
+    for (int i = 0; i < rows.size(); i++) {
+      checkArray(records[i + 1], rows.get(i));
+    }
+  }
+
+  private void checkArray(String[] values, List<String> expected) {
+    for (int i = 0; i < expected.size(); i++) {
+      assertEquals(expected.get(i), values[i]);
+    }
+  }
```

Comment:
Rename `values` -> `actual` and switch places with `expected` so it's consistent with `assertEquals`.